chore: skip orchestrator info requests if service uri not set by orchestrator

ad-astra-video opened the pull request:
**What does this pull request do? Explain your changes. (required)**
Update `db_discovery.go` to skip starting orchestrator http client if the service uri is not set by the orchestrator. At time of this commit, 71 orchestrators have not registered for the AI subnet so this skips a lot of failed http connections.

Docker build is at `docker pull adastravideo/go-livepeer:ai-video-test`

**Specific updates (required)**
- add a check to confirm a `Host` exists in the parsed service uri.

**How did you test each of these updates (required)**
Built docker file and started up gateway
[startup.log](https://github.com/user-attachments/files/16681763/startup.log)


**Checklist:**
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] Read the [contribution guide](./CONTRIBUTING.md)
- [X] `make` runs successfully
- [ ] All tests in `./test.sh` pass
- [ ] README and other documentation updated
- [ ] [Pending changelog](./CHANGELOG_PENDING.md) updated

ad-astra-video committed:
skip orchestrator info request if orch has no service uri set

rickstaa commented:
@ad-astra-video Do we still need this now that we've merged into the master branch? Are there any orchestrators that still have their service URI set to an empty string ("") by mistake? I noticed this issue once before, but I'm not sure if any orchestrators are still affected.

ad-astra-video commented:
The AI service registry is still not fully used by Orchestrators.  This is just a thing to ease development starting gateways faster.  For production gateways it is not likely noticed much.

I have not issue closing this but it also doesn't make sense to try to connect to a URL that is blank.

ad-astra-video commented:
Counted the empty entries on last test gateway startup.  There are still 59 empty urls in the `AIServiceRegistry`.

rickstaa committed:
refactor: minor comment improvements

This commit makes minor updates to comments for better clarity and consistency. No
functional code changes are included.

rickstaa committed:
refactor: simplify orch discovery skip logic

This commit refactors the orchestration skip logic by replacing the use of a new
data channel with a direct error throw.

rickstaa reviewed (approved):
@ad-astra-video I’ve simplified the logic in [this commit](https://github.com/livepeer/go-livepeer/pull/3149/commits/9c08eeb4cee9cb0d0f65d4f31135c5da086ad8a4). If you’re aligned with this change, we can proceed with the merge.

rickstaa commented on discovery/db_discovery.go:
```diff
@@ -278,7 +278,7 @@ func (dbo *DBOrchestratorPoolCache) cacheDBOrchs() error {
 		// Do not connect if URI host is not set
 		if uri.Host == "" {
-			skipped <- 1
+			errc <- fmt.Errorf("skipping orch=%v, URI not set", dbOrch.EthereumAddr)
```
@victorges Should we throw this error, or would it be cleaner to simply display the fetched orchestrators? See @ad-astra-video's first solution (i.e. https://github.com/livepeer/go-livepeer/pull/3149/commits/6667483959e6b4ed2520a36b903f3728d0753ac7).

leszko committed:
Merge branch 'master' into ai-video-skip-no-service-uri-orchs
